From 4377df8ca0573bcd4839ed09fd37f7a4105b576a Mon Sep 17 00:00:00 2001
From: Lawrence Rust <lvr@softsystem.co.uk>
Date: Tue, 15 Mar 2011 12:34:32 +0100
Subject: [PATCH 018/124] MediaMonitor::GetMedias - search for multile media
 types

Add the ability to request multiple media types when searching  MediaMonitor
disks.

This is useful in the MythGallery plugin which registers specific image
filetypes with MediaMonitor.  When a CD/USB is inserted with media
containing images, the disk is searched statistically and the media type
is set to MEDIATYPE_MGALLERY.  However, when MythGallery shows a list of
devices with supported media it needs to ask for both MEDIATYPE_MGALLERY
and MEDIATYPE_DATA.  MythGallery is the only caller of the
MediaMonitor::GetMedias function.

Ticket URL: <http://code.mythtv.org/trac/ticket/9520>

Signed-off-by: Lawrence Rust <lvr@softsystem.co.uk>
---
 mythplugins/mythgallery/mythgallery/iconview.cpp |    2 +-
 mythtv/libs/libmyth/mythmediamonitor.cpp         |   10 +++++-----
 mythtv/libs/libmyth/mythmediamonitor.h           |    2 +-
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/mythplugins/mythgallery/mythgallery/iconview.cpp b/mythplugins/mythgallery/mythgallery/iconview.cpp
index 18b3bf4..c4f2987 100644
--- a/mythplugins/mythgallery/mythgallery/iconview.cpp
+++ b/mythplugins/mythgallery/mythgallery/iconview.cpp
@@ -618,7 +618,7 @@ bool IconView::HandleMediaEscape(MediaMonitor *mon)
 {
     bool handled = false;
     QDir curdir(m_currDir);
-    QList<MythMediaDevice*> removables = mon->GetMedias(MEDIATYPE_DATA);
+    QList<MythMediaDevice*> removables = mon->GetMedias(MEDIATYPE_DATA|MEDIATYPE_MGALLERY);
     QList<MythMediaDevice*>::iterator it = removables.begin();
     for (; !handled && (it != removables.end()); ++it)
     {
diff --git a/mythtv/libs/libmyth/mythmediamonitor.cpp b/mythtv/libs/libmyth/mythmediamonitor.cpp
index 9eff7db..853095d 100644
--- a/mythtv/libs/libmyth/mythmediamonitor.cpp
+++ b/mythtv/libs/libmyth/mythmediamonitor.cpp
@@ -454,7 +454,7 @@ void MediaMonitor::StopMonitoring(void)
  *
  *   NOTE: This function can block.
  *
- *  \sa Unlock(MythMediaDevice *pMedia), GetMedias(MythMediaType mediatype)
+ *  \sa Unlock(MythMediaDevice *pMedia), GetMedias(unsigned mediatypes)
  */
 bool MediaMonitor::ValidateAndLock(MythMediaDevice *pMedia)
 {
@@ -471,7 +471,7 @@ bool MediaMonitor::ValidateAndLock(MythMediaDevice *pMedia)
 /** \fn MediaMonitor::Unlock(MythMediaDevice *pMedia)
  *  \brief decrements the MythMediaDevices reference count
  *
- *  \sa ValidateAndLock(MythMediaDevice *pMedia), GetMedias(MythMediaType mediatype)
+ *  \sa ValidateAndLock(MythMediaDevice *pMedia), GetMedias(unsigned mediatypes)
  */
 void MediaMonitor::Unlock(MythMediaDevice *pMedia)
 {
@@ -550,7 +550,7 @@ QString MediaMonitor::GetMountPath(const QString& devPath)
     return mountPath;
 }
 
-/** \fn MediaMonitor::GetMedias(MythMediaType mediatype)
+/** \fn MediaMonitor::GetMedias(unsigned mediatypes)
  *  \brief Ask for available media. Must be locked with ValidateAndLock().
  *
  *   This method returns a list of MythMediaDevice pointers which match
@@ -568,7 +568,7 @@ QString MediaMonitor::GetMountPath(const QString& devPath)
  *  \sa ValidateAndLock(MythMediaDevice *pMedia)
  *  \sa Unlock(MythMediaDevice *pMedia)
  */
-QList<MythMediaDevice*> MediaMonitor::GetMedias(MythMediaType mediatype)
+QList<MythMediaDevice*> MediaMonitor::GetMedias(unsigned mediatypes)
 {
     QMutexLocker locker(&m_DevicesLock);
 
@@ -577,7 +577,7 @@ QList<MythMediaDevice*> MediaMonitor::GetMedias(MythMediaType mediatype)
     QList<MythMediaDevice*>::iterator it = m_Devices.begin();
     for (;it != m_Devices.end(); ++it)
     {
-        if (((*it)->getMediaType() & mediatype) &&
+        if (((*it)->getMediaType() & mediatypes) &&
             (((*it)->getStatus() == MEDIASTAT_USEABLE) ||
              ((*it)->getStatus() == MEDIASTAT_MOUNTED) ||
              ((*it)->getStatus() == MEDIASTAT_NOTMOUNTED)))
diff --git a/mythtv/libs/libmyth/mythmediamonitor.h b/mythtv/libs/libmyth/mythmediamonitor.h
index b723bbc..a859d22 100644
--- a/mythtv/libs/libmyth/mythmediamonitor.h
+++ b/mythtv/libs/libmyth/mythmediamonitor.h
@@ -58,7 +58,7 @@ class MPUBLIC MediaMonitor : public QObject
     // first validate the pointer with ValidateAndLock(), if true is returned
     // it is safe to dereference the pointer. When finished call Unlock()
     QList<MythMediaDevice*> GetRemovable(bool mounted=false);
-    QList<MythMediaDevice*> GetMedias(MythMediaType mediatype);
+    QList<MythMediaDevice*> GetMedias(unsigned mediatypes);
     MythMediaDevice*        GetMedia(const QString &path);
 
     void MonitorRegisterExtensions(uint mediaType, const QString &extensions);
-- 
1.7.9.5

